lib/html_quote.cc

```diff
@@ -7,47 +7,25 @@
  */
 
 #include "squid.h"
+#include "base/CharacterSet.h"
 #include "html_quote.h"
 
-#if HAVE_STRING_H
-#include <string.h>
-#endif
+#include <cstring>
+#include <unordered_map>
 
-/*
- *  HTML defines these characters as special entities that should be quoted.
- */
-static struct {
-    unsigned char code;
-    const char *quote;
-} htmlstandardentities[] =
 
-{
-    /* NOTE: The quoted form MUST not be larger than 6 character.
-     * see close to the MemPool commend below
-     */
-    {
-        '<', "&lt;"
-    },
-    {
-        '>', "&gt;"
-    },
-    {
-        '"', "&quot;"
-    },
-    {
-        '&', "&amp;"
-    },
-    {
-        '\'', "&#39;"
-    },
-    {
-        0, NULL
-    }
+const std::unordered_map<char,const char *> htmlEntities = {
+    {'<', "&lt;"},
+    {'>', "&gt;"},
+    {'"', "&quot;"},
+    {'&', "&amp;"},
+    {'\'', "&#39;"}
```

Comment:
It would be good to add a C++ comment that documents what this non-entity is doing in the entities map (or in the refactored code). In other words, why this is not `&apos;`... I do not insist on this change.

Reply:
There is actually no reason why it can't be &apos; that I know of.
In fact, why don't we enrich this table with the ASCII part of the list from https://www.w3schools.com/html/html_entities.asp ?